apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -19,39 +23,151 @@ enum ParseError {
     #[error("Invalid action received: {0}")]
     InvalidAction(u8),
     #[error("Error occurred when sending keypress event")]
-    KeypressError(),
+    KeypressError(#[from] uinput::Error),
     #[error("Button value invalid: {0}")]
-    Button(#[from] ButtonError),
+    InvalidButton(u8),
+    #[error("Button unsupported: {0:?}")]
+    UnsupportedButton(Button),
```

Comment:
There seemed to be a distinction between "invalid" and "not yet supported". I'm not sure this error type is the right place for this, and maybe it becomes something other than `ParseError` if it stays here? 🤷